htlcswitch/switch.go

```diff
@@ -1952,6 +1960,16 @@ func (s *Switch) addLiveLink(link ChannelLink) {
 		s.interfaceIndex[peerPub] = make(map[lnwire.ChannelID]ChannelLink)
 	}
 	s.interfaceIndex[peerPub][link.ChanID()] = link
+
+	// Since the link addition has succeeded, this channel is now officially
+	// active and we can tell the ChannelNotifier about it.
+	channel, err := s.cfg.DB.FetchOpenChannel(peerPub[:], link.ChannelPoint())
```

Comment:
Not sure if there are performance considerations we should take doing DB queries when adding and removing links. Will let @cfromknecht comment on this . :)